### PR TITLE
[Gen 2] Add the missing manage attributes page and update the user profile page for custom attributes.

### PR DESCRIPTION
**Pull request by salihgueler**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] Swift
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx [hunk 1]
```diff
@@ -164,8 +375,213 @@ The backend authentication resource authored using `defineAuth` does not current
...
+    username: username,
+    password: password,
+    options: SignUpOptions(
+      userAttributes: userAttributes,
+    ),
+  );
+}
+```
+
+When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Although Cognito prepends a "custom:" prefix on the attribute name, there is no need for you to add this in Amplify Flutter's custom attribute constructor.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx:406]: can we show/link to the cdk snippet for custom attributes?

File: src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx [hunk 2]
```diff
@@ -164,8 +375,213 @@ The backend authentication resource authored using `defineAuth` does not current
...
+    .subscribe(
+        result -> Log.i("AuthQuickstart", result.toString()),
+        error -> Log.e("AuthQuickstart", error.toString())
+    );
+```
+
+</Block>
+</BlockSwitcher>
+
+When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Please keep in mind, Amazon Cognito prepends a "custom:" prefix on the attribute name.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx:492]: same here

File: src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx [hunk 3]
```diff
@@ -164,8 +375,213 @@ The backend authentication resource authored using `defineAuth` does not current
...
+}
+```
+
+</Block>
+
+</BlockSwitcher>
+
+</InlineFilter>
+
+When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Please keep in mind, Amazon Cognito prepends a "custom:" prefix on the attribute name.When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Please keep in mind, Amazon Cognito prepends a "custom:" prefix on the attribute name.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/manage-user-profile/index.mdx:580]: can we link to the custom attributes snippet example?

File: src/pages/[platform]/build-a-backend/auth/manage-attributes/index.mdx [hunk 4]
```diff
@@ -0,0 +1,645 @@
...
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+<InlineFilter filters={['flutter']}>
+## Fetch the current user's attributes
+
+Invoke the following api to get the list of attributes assigned to the user.
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/manage-attributes/index.mdx:25]: API

**Review by aspittel — DISMISSED** [review]

**Review by renebrandel — DISMISSED** [review]

**Review by renebrandel — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/auth/manage-attributes/index.mdx [hunk 5]
```diff
@@ -0,0 +1,645 @@
...
+    username: username,
+    password: password,
+    options: SignUpOptions(
+      userAttributes: userAttributes,
+    ),
+  );
+}
+```
+
+When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Although Cognito prepends a "custom:" prefix on the attribute name, there is no need for you to add this in Amplify Flutter's custom attribute constructor.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/manage-attributes/index.mdx:183]: can we link to the custom attributes example in the auth docs? https://next-docs.amplify.aws/react/build-a-backend/auth/override-cognito/#custom-attributes

**Review by josefaidt — APPROVED** [review]

**Pull request merged**